File: sum_of_three/sum.py
def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        if 1 <= n <= 6 or n == 9:
            print("NO")
        elif n % 3:
            print("YES")
            print(1, 2, n - 3)
        else:
            print("YES")
            print(1, 4, n - 5)
    
    
if __name__ == "__main__":
    main()

File: sum_of_three/sum.c
#include <stdio.h>

int main(void)
{
    int t;
    scanf("%d", &t);
    for (int i = 0; i < t; i++)
    {
        int n;
        scanf("%d", &n);
        if ((n >= 1 && n <= 6) || n == 9)
        {
            printf("NO\n");
        }

        else
        {
            printf("YES\n");
            if (((n - 3) % 3))
            {
                printf("%d %d %d\n", 1, 2, n - 3);
            }

            else
            {
                printf("%d %d %d\n", 1, 4, n - 5);
            }
        }
    }
}
